minari/data_collector/data_collector.py

```diff
@@ -348,76 +348,60 @@ def clear_buffer(dictionary_buffer: EpisodeBuffer, episode_group: h5py.Group):
-                elif all([isinstance(entry, OrderedDict) for entry in data]):
-
+                elif all(map(lambda elem: isinstance(elem, OrderedDict), data)):
                     # we have a list of OrderedDicts, so we need to act appropriately
                     dict_data = {
                         key: [entry[key] for entry in data]
                         for key, value in data[0].items()
                     }
-
-                    if key in episode_group:
-                        eps_group_to_clear = episode_group[key]
-                    else:
-                        eps_group_to_clear = episode_group.create_group(key)
+                    eps_group_to_clear = episode_group.get(
+                        key, episode_group.create_group(key)
+                    )
                     clear_buffer(dict_data, eps_group_to_clear)
                 else:
-                    # convert data to numpy
-                    np_data = np.asarray(data)
-                    assert np.all(
-                        np.logical_not(np.isnan(np_data))
-                    ), "Nan found after cast to nump array, check the type of 'data'."
+                    if all(map(lambda elem: isinstance(elem, str), data)):
+                        data_shape = (len(data),)
+                        dtype = h5py.string_dtype(encoding="utf-8")
+                    else:
+                        data = np.asarray(data)
+                        data_shape = data.shape
+                        dtype = data.dtype
+                        assert np.all(
+                            np.logical_not(np.isnan(data))
+                        ), "Nan found after cast to nump array, check the type of 'data'."
 
-                    # Check if last episode group is terminated or truncated
                     if (
                         not self._last_episode_group_term_or_trunc
                         and key in episode_group
                     ):
-                        # Append to last episode group datasets
-                        if key not in STEP_DATA_KEYS and key != "infos":
-                            # check current dataset size directly from hdf5 since
-                            # non step data (actions, obs, rew, term, trunc) may not be
-                            # added in a per-step/sequential basis, including "infos"
-                            current_dataset_shape = episode_group[key].shape[0]
-                        else:
-                            current_dataset_shape = self._last_episode_n_steps
-                            if key == "observations":
-                                current_dataset_shape += (
-                                    1  # include initial observation
-                                )
+                        current_dataset_shape = episode_group[key].shape[0]
+                        if key == "observations":
+                            current_dataset_shape += 1
```

Comment:
@rodrigodelazcano I tried refactoring this else branch

Important changes:
 - the length of the episode is always taken from `episode_group[key].shape[0]` without using `self._last_episode_n_steps`. Shouldn't they always be the same?
- Always specify the `maxshape` argue ven when you dump everything (not mutable). It allows to merge the two branches
- Add `dtype` specification on create_dataset. It uses the dtype of the ndarray

Reply:
1. Yes this is better. They should always be the same. The initial intention of `self._last_episode_n_steps` was to not read this value directly from file. But I agree this doesn't make any sense if we are doing it for other cases so I approve the change. However, with these changes you should remove the condition for when the key is `observations` is not required since the +1 will  be already present in the hdf5 `observations` dataset.
2. Agree
3. Very necessary. Agree